### PR TITLE
Compiler tests cleanup, part 2

### PR DESCRIPTION
**Pull request by psfinaki**
Continuation of this: https://github.com/dotnet/fsharp/pull/15690

### CODE REVIEW

File: tests/FSharp.Compiler.UnitTests/PerfTests.fs [hunk 1]
```diff
@@ -51,54 +47,54 @@ let ``Test request for parse and check doesn't check whole project`` () =
...
     printfn "checking no extra background typechecks...., backgroundCheckCount.Value = %d" backgroundCheckCount.Value
-    (backgroundCheckCount.Value  <= 10) |> shouldEqual true // only two extra typechecks of files
+    (backgroundCheckCount.Value  <= 10) |> Assert.shouldBeTrue // only two extra typechecks of files
 
     printfn "checking (pD - pC) = %d" (pD - pC)
-    (pD - pC) |> shouldEqual 0
+    (pD - pC) |> Assert.shouldBe 0
     printfn "checking (tD - tC) = %d" (tD - tC)
-    (tD - tC) |> shouldEqual 1
+    (tD - tC) |> Assert.shouldBe 1
```
Comment (T-Gro) [inline tests/FSharp.Compiler.UnitTests/PerfTests.fs:74]: I don't think there is any  value in changing all the trivial 'shouldEqual` occurrences altogether within these PRs, especially when the "previous" code was going via a helper function.

Just makes the diff and assessment of changes unnecessarily large, and easily leads to bugs.

(on example of a possible mistake: the existing 'shouldEqual' was written in  way to provide type safety and only compare elements of equal types, which is important especially with type inference. The Assert.shouldBe which was used instead allows arguments of different types, which can lead to inferring a more generic type (obj,IComparable,..) which then behaves differently.)

Firstly I do not think it matters that much when run, assertions should be possible to be combined from multiple libraries and it works.

If it does matter somewhere significantly (??), the function could be instead temporarily shadowed (and added via an 'open'), and then eventually the `shouldEqual` reimplemented to make it fit.

Comment (psfinaki) [reply tests/FSharp.Compiler.UnitTests/PerfTests.fs]: Yes, this is a valid point, I didn't notice typed vs untyped comparison.

Well, we cannot keep FsUnit as it builds on top of NUnit which we are migrating from. So instead I added the typed comparison to the `Assert` file and opened it where needed hence minimizing the changes. I imagine that's what you meant by shadowing?

Comment (T-Gro) [reply tests/FSharp.Compiler.UnitTests/PerfTests.fs]: Yep, that's good.

(also, I do not think we have to move away from FsUnit - usage of assertion library should be orthogonal choice to the usage of test runner. And, the inner workings of FsUnit.fs could be moved to use XUnit if that does make a difference)

**Review by T-Gro — APPROVED** [review]

File: tests/FSharp.Compiler.UnitTests/PerfTests.fs (hunk 1 above)
Comment (psfinaki) [reply tests/FSharp.Compiler.UnitTests/PerfTests.fs]: That's also fair, eventually we can stay with FsUnit, again it's just about consolidating all these helpers we have here and there.

**Review by 0101 — APPROVED** [review]

**Pull request merged**